src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,384 @@
+            result=False,
+        )
+
+    for sddc in sddc_list:
+        if sddc["name"] == sddc_name:
+            return vmc_state._create_state_response(
+                name=name, comment="SDDC is already present", result=True
+            )
+
+    if __opts__.get("test"):
+        log.info("vmc_sddc present is called with test option")
+        return vmc_state._create_state_response(
+            name=name,
+            comment="SDDC {} would have been created".format(sddc_name),
+        )
+
+    created_sddc = __salt__["vmc_sddc.create"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_name=sddc_name,
+        num_hosts=num_hosts,
+        provider=provider,
+        region=region,
+        account_link_config=account_link_config,
+        account_link_sddc_config=account_link_sddc_config,
+        deployment_type=deployment_type,
+        host_instance_type=host_instance_type,
+        msft_license_config=msft_license_config,
+        sddc_id=sddc_id,
+        sddc_template_id=sddc_template_id,
+        sddc_type=sddc_type,
+        size=size,
+        skip_creating_vxlan=skip_creating_vxlan,
+        sso_domain=sso_domain,
+        storage_capacity=storage_capacity,
+        vpc_cidr=vpc_cidr,
+        vxlan_subnet=vxlan_subnet,
+        validate_only=validate_only,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in created_sddc:
+        return vmc_state._create_state_response(
+            name=name, comment="Failed to add SDDC : {}".format(created_sddc["error"]), result=False
+        )
+
+    return vmc_state._create_state_response(
+        name=name,
+        comment="Created SDDC {}".format(sddc_name),
+        new_state=created_sddc,
+        result=True,
+    )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    force_delete=False,
+    retain_configuration=False,
+    template_name=None,
+    verify_ssl=True,
+    cert=None,
+):
+
+    """
+    Ensure a given SDDC does not exist for the given organization.
+
+    name
+        Indicates the SDDC ID, UUID identifying the SDDC.
+
+    hostname
+        The host name of VMC.
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations.
+
+    authorization_host
+        Hostname of the VMC cloud console.
+
+    org_id
+        The ID of organization to which the SDDC belongs to.
+
+    force_delete: Boolean
+        (Optional) If true, will delete forcefully.
+        Beware: do not use the force flag if there is a chance an active provisioning or deleting task is running
+        against this SDDC. This option is restricted.
+
+    retain_configuration: Boolean
+        (Optional) If = 'true', the SDDC's configuration is retained as a template for later use.
+        This flag is applicable only to SDDCs in ACTIVE state.
```

Comment:
**comment** the upstream documentation here is not a very good model. Do we know if force_delete just doesn't work if there's an active provisioning task running? It seems like that's *probably* what was meant, e.g.

```
If True, delete this SDDC despite XYZ. If an active provisioning or deleting task is running against this SDDC then this will fail.
```

Or something to that effect. I don't know what they really are after here. 

**suggestion** Should be `If True`  on retain_configuration, too.

Reply:
This delete api without force will fail if there are active provisioning
 force_delete will delete even if there are active provisioning, and it destorys to all and that's the reason it is in restricted action.